dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/decorator/HttpServerDecorator.java

```diff
@@ -155,6 +166,16 @@ public AgentSpan startSpan(
     return span;
   }
 
+  public AgentSpan startSpanFromContext(
+      String instrumentationName, REQUEST_CARRIER carrier, Context context) {
+    return startSpan(instrumentationName, carrier, getExtractedSpanContext(context));
+  }
+
+  public AgentSpanContext.Extracted getExtractedSpanContext(Context context) {
```

Comment:
nitpick: you don't have to suppose the context will always come from extraction -- this might be true later when we keep migrating the helper
```suggestion
  public AgentSpanContext.Extracted getSpanContext(Context context) {
```